### ISSUE TITLE
PortalListener causing performance issues

### ISSUE THREAD

**Issue by bloodmc** [trentech/ProjectPortals#9]
Just ran a fresh timings and notice Project Portals was causing significant lag in its PortalListener class.

Project Portals::PortalListener
count(489602) total(23.12% 19.369s, 18.96% of tick)  
avg(0.04ms per - 9.48ms/239.41 per tick

This only occured as leaves were ticking. However, I have 'leaf-decay' disabled so this may be an issue when this is turned off with your plugin. Let me know if it something on our end.

Another performance issue, this time with entities. The following occured during chicken ticks

Project Portals::PortalListener
count(126328) total(10.96% 5.98s, 2.80% of tick)  
avg(0.05ms per - 1.40ms/29.56 per tick)

I would suggest putting timings in each of your methods for all listener classes so you can fix these performance issues. Take a look at how PEX implemented timings support here

https://github.com/PEXPlugins/PermissionsEx/commit/2f7d8fdf42106c3c703c6b04c9632fbc31b2ae0e

**Comment by trentech** [comment]
OK, I just pushed some Timings stuff for most the listener classes. Haven't spent a lot of time with timings but it looks like it's working. It's hard to test performance issues when I'm the only one on the server so hopefully this will help. The leaf decay might no longer be an issue soon (at least I hope) due to the new PortalBuilder class I started experimenting with. I also made some small changes to handling varies move events but not sure if it'll make a difference yet.
